assert/assertions.go

```diff
@@ -1167,25 +1168,44 @@ func Panic(t testing.TB, fn func(), opts ...Option) {
 // The function parameter must not be nil.
 func NotPanic(t testing.TB, fn func(), opts ...Option) {
 	t.Helper()
-	panicked, r := didPanic(fn)
+	cfg := processOptions(opts...)
+	panicked, r, stack := didPanic(fn, cfg.StackTrace)
 	if panicked {
-		cfg := processOptions(opts...)
-		errorMsg := fmt.Sprintf("Expected for the function to not panic, but it panicked with: %v", r)
+		var messageBuilder strings.Builder
+		messageBuilder.WriteString("Expected for the function to not panic, but it panicked with: ")
+		messageBuilder.WriteString(fmt.Sprintf("%v", r))
+
+		if stack != "" {
+			messageBuilder.WriteString("\nStack trace:\n")
+			messageBuilder.WriteString(stack)
+		}
+
+		if !cfg.StackTrace {
+			messageBuilder.WriteString("\nNote: Stack trace is not available when should.WithStackTrace() is not used")
+		}
+
 		if cfg.Message != "" {
-			fail(t, "%s\n%s", cfg.Message, errorMsg)
+			fail(t, "%s\n%s", cfg.Message, messageBuilder.String())
 			return
 		}
 
-		fail(t, errorMsg)
+		fail(t, messageBuilder.String())
 	}
 }
 
 // didPanic executes a function and reports whether it panicked, returning the recovered value.
-func didPanic(fn func()) (panicked bool, recovered any) {
+func didPanic(fn func(), withStackTrace bool) (panicked bool, recovered any, stack string) {
```

Comment:
Please consider returning a struct such as `panicInfo`

Having 3 returned value make it uneasy.